### DISCUSSION TITLE
logger

### DISCUSSION THREAD

**Discussion by paskuale75** [simialbi/yii2-elfinder#12] [Q&A]
There's a way to record the actions performed on the folder, in the logger ?
example:
`Yii::info($_username . ' changed the contents of the folder : ' . $model->alias, 'elfinder');`

**Comment by simialbi** [comment]
Would be a possible enhancement

**Reply by paskuale75** [reply]
Perfect, I offer you my modest support if you need it

**Reply by paskuale75** [reply]
@simialbi have you some advice ? Where to start? I could try my hand I have some free time these days.

**Reply by simialbi** [reply]
There is an event triggered by each action:
https://github.com/simialbi/yii2-elfinder/blob/master/src/ElFinder.php#L495

You can bind a listener to this events and trigger the log.

**Reply by simialbi** [reply]
@paskuale75 did you try this approach?

**Reply by paskuale75** [reply]
Hi @simialbi 
To tell the truth I haven't tried it yet, because I'm using a small project. But soon I'm going to rely on your advice. I take this opportunity to ask you one more thing, can I use the accessControls on the elFinder extension, to decide which role can or cannot access the folder?

**Reply by paskuale75** [reply]
Hi @simialbi works like a charme !!!!! 
I added this in my module init() 
`use simialbi\yii2\elfinder\ElFinder;` in use section...

`$this->on(ElFinder::EVENT_AFTER_UPLOAD, Yii::info(' ...has added one file... the name of file is <NAME_OF_FILE>' , 'app'));`

1) Just a problem, I don't know why but the log writes it to me twice !
2) Who knows if there is a way to also have the details on what has been uploaded  ! ^_^

**Reply by paskuale75** [reply]
> There is an event triggered by each action:
> https://github.com/simialbi/yii2-elfinder/blob/master/src/ElFinder.php#L495
> 
> You can bind a listener to this events and trigger the log.

bind a listener ? Mmmm any example ? where should i do it ? Thanks

**Comment by paskuale75** [comment]
I'm very happy with this [extension](https://github.com/apolloeleven/yii2-logger) for manage log.
